scarb/src/compiler/db.rs

```diff
@@ -169,6 +169,7 @@ fn build_project_config(unit: &CairoCompilationUnit) -> Result<ProjectConfig> {
                         .any(|target| {
                             target.group_id.clone().unwrap_or(target.name.clone())
                                 == component.package.id.name.to_smol_str()
+                            && component_as_dependency.cairo_package_name() != component.cairo_package_name()
```

Comment:
Why this change?

Reply:
To make sure, we do not add dependency on self here.